doc/design/var_desc.md

```diff
@@ -0,0 +1,122 @@
+## Background
+PaddlePaddle divides the description of neural network computation graph into two stages: compile time and runtime.
+
+PaddlePaddle use proto message to describe compile time graph for
+
+1. Computation graph should be able to be saved to a file.
+1. In distributed training, the graph will be serialized and send to multiple workers.
+
+The computation graph is constructed by Data Node and Operation Node. The concept to represent them is in the table below.
```

Comment:
What are the Node and Edge of computation graph?

Reply:
data and operator are both Node, the Edge is the relationship between data and operator: input/output relation.